internal/state/range_test.go

```diff
@@ -142,29 +142,29 @@ func TestNewRangeTracker(t *testing.T) {
 	}
 }
 
-func TestMaxOffsetTracker_Insert(t *testing.T) {
+func TestSingleRangeTracker_Insert(t *testing.T) {
 	t.Run("NonOverlapping", func(t *testing.T) {
 		t.Parallel()
-		tracker := newMaxOffsetTracker("test")
+		tracker := newSingleRangeTracker("test")
 		assert.True(t, tracker.Insert(Range{start: 0, end: 5}))
 		assert.True(t, tracker.Insert(Range{start: 20, end: 30}))
 		assert.Equal(t, 1, tracker.Len())
-		assert.Equal(t, "[0-30]", tracker.String())
+		assert.Equal(t, "[0-30]", tracker.Ranges().String())
```

Comment:
is this the same expected behavior? looks like we are assuming the missing log portion inbetween those 2 ranges will be pushed eventually? I think the previous behavior is the offset doesn't get written unless it's processed for sure and sequentially.